infrastructure/src/main/java/org/corfudb/infrastructure/SequencerServer.java

```diff
@@ -575,6 +628,38 @@ private void handleAllocation(CorfuPayloadMsg<TokenRequest> msg, ChannelHandlerC
                 new TokenResponse(token, backPointerMap.build())));
     }
 
+    /**
+     * This method handles the request of streams address maps.
+     * It computes the addresses requested for each stream
+     * and returns this map under a StreamsAddressResponse message.
+     */
+    @ServerHandler(type = CorfuMsgType.STREAMS_ADDRESS_REQUEST)
+    private synchronized void handleStreamsAddressRequest(CorfuPayloadMsg<StreamsAddressRequest> msg,
+                                             ChannelHandlerContext ctx, IServerRouter r) {
+        List<StreamAddressRange> streamsRangesRequested = msg.getPayload().getStreamsRanges();
+        Map<UUID, StreamAddressSpace> streamsAddressSpaceMapToReturn = new HashMap<>();
+
+        Roaring64NavigableMap addressMap;
+
+        for (StreamAddressRange streamAddressRange : streamsRangesRequested) {
+            UUID streamId = streamAddressRange.getStreamID();
+            try {
+                // Get all addresses in the requested range
+                addressMap = streamsAddressSpaceMap.get(streamId).getAddressesInRange(streamAddressRange.start, streamAddressRange.end);
+                Long trimMark = streamsAddressSpaceMap.get(streamId).getTrimMark();
+                streamsAddressSpaceMapToReturn.put(streamId, new StreamAddressSpace(trimMark, addressMap));
+            } catch (NullPointerException ne) {
```

Comment:
Something wrong with the code if it catches NPE, please refactor

Reply:
done.